py3status/py3.py

```diff
@@ -608,28 +609,39 @@ def format_contains(self, format_string, name):
         will fail if the format string contains placeholder formatting
         eg ``'{placeholder:.2f}'``
         """
+        if isinstance(names, basestring):
+            names = [names]
 
         # We cache things to prevent parsing the format_string more than needed
-        try:
-            return self._format_placeholders_cache[format_string][name]
-        except KeyError:
-            pass
+        boolean_list = []
+        for name in names:
+            try:
+                boolean_list.append(
+                    self._format_placeholders_cache[format_string][name])
+            except KeyError:
+                pass
+        if len(names) == len(boolean_list):
+            return any(boolean_list)
 
         if format_string not in self._format_placeholders:
             placeholders = self._formatter.get_placeholders(format_string)
             self._format_placeholders[format_string] = placeholders
         else:
             placeholders = self._format_placeholders[format_string]
 
-        result = False
-        for placeholder in placeholders:
-            if fnmatch(placeholder, name):
-                result = True
-                break
         if format_string not in self._format_placeholders_cache:
             self._format_placeholders_cache[format_string] = {}
-        self._format_placeholders_cache[format_string][name] = result
-        return result
+
+        boolean_list = []
+        for name in names:
+            result = False
+            for placeholder in placeholders:
+                if fnmatch(placeholder, name):
+                    result = True
+                    break
+            boolean_list.append(result)
+            self._format_placeholders_cache[format_string][name] = result
+        return any(boolean_list)
```

Comment:
this can be simplified. 


we only need one match to cache the result and return, any extra work is wasted effort